File: app/src/main/java/com/danielmaia/desafioandroid/dao/RepoDao.java
package com.danielmaia.desafioandroid.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.danielmaia.desafioandroid.model.Repo;

import java.util.List;

@Dao
public interface RepoDao {

    @Query("SELECT * FROM Repo")
    List<Repo> getRepos();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(Repo repo);

    @Query("SELECT * FROM Repo WHERE id = :id")
    List<Repo> getById(int id);

    @Query("DELETE FROM Repo")
    void deleteAll();

    @Query("DELETE FROM Repo WHERE id = :id")
    void deleteById(long id);
}

File: app/src/main/java/com/example/desafioandroid/data/database/dao/RepoDao.kt
package com.example.desafioandroid.data.database.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.example.desafioandroid.data.database.entities.RepoEntity

@Dao
interface RepoDao {

    @Query("SELECT * FROM repo_table")
    suspend fun getAllRepos(): List<RepoEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertRepos(repo: List<RepoEntity>)

    @Query("DELETE FROM repo_table")
    suspend fun deleteAllRepos()
}